ASTInfoGenerator/DeclParser/DeclParser.py

```diff
@@ -0,0 +1,382 @@
+        record_decl_chunks = record_decl.split(" ")
+        struct_name = record_decl_chunks[len(record_decl_chunks) - 2]
+
+        return RecordDecl(struct_name)
+
+    def _parse_parm_var_decl(self, parm_var_decl: str, param_index: int):
+        quote_second_index = parm_var_decl.find(
+            "'", parm_var_decl.find("'") + 1)
+
+        param_type = parm_var_decl[
+            parm_var_decl.find("'") + 1: quote_second_index
+        ]
+
+        # check if param type is formatted like 'typedef_name':'struct_name *'
+        # we extract it into just `struct_name *`
+        start_of_struct_type_name_index = quote_second_index + 3
+        if len(parm_var_decl) > start_of_struct_type_name_index and "*" in parm_var_decl[start_of_struct_type_name_index:]:
+            param_type = parm_var_decl[start_of_struct_type_name_index: len(
+                parm_var_decl) - 1]
+            param_type = param_type[: param_type.find("'")]
+
+        param_type = param_type.replace(" ", "")
+        logout(f"param added {param_type} for {parm_var_decl}")
+
+        return ParmVarDecl(param_index, param_type)
+
+    def _parse_typedef_decl(self, typedef_decl: str):
+        if not ("referenced" in typedef_decl and "struct" in typedef_decl):
+            return None
+        original_name = ""
+        start_index = typedef_decl.find("'") + 1
+        cur_char = typedef_decl[start_index]
+
+        while (cur_char != "'"):
+            original_name += cur_char
+
+            start_index += 1
+            cur_char = typedef_decl[start_index]
+
+        original_name_split = original_name.split(" ")
+        if len(original_name_split) > 1 and original_name_split[1] != "":
+            original_name = original_name_split[1]
+        else:
+            original_name = original_name_split[0]
+
+        typedef_alias_name = ""
+        start_index = typedef_decl.find("referenced") + len("referenced") + 1
+        cur_char = typedef_decl[start_index]
+
+        while (cur_char != " "):
+            typedef_alias_name += cur_char
+
+            start_index += 1
+            cur_char = typedef_decl[start_index]
+
+        if original_name != typedef_alias_name:
+            return TypedefDecl(original_name, typedef_alias_name)
+
+        return None
+
+    def _parse_field_decl(self, field_decl: str, field_index: int):
+        start_index = field_decl.find("'") - 2
+        cur_char = field_decl[start_index]
+        field_name = ""
+
+        while (cur_char != " "):
+            field_name = cur_char + field_name
+
+            start_index -= 1
+            cur_char = field_decl[start_index]
+
+        return FieldDecl(field_name, field_index)
+
+    # only checks for vars whose type is struct
+    def _parse_var_decl(self, var_decl: str, specifier_manager: SpecifierManager):
+        var_decl_chunks = var_decl.split(" ")
+
+        logout(var_decl)
+        logout(var_decl_chunks)
+
+        if var_decl_chunks[5] == "used":
```

Comment:
There is some kind of assumption being made here that I don't understand. What's the significance of "used"? I think a comment is needed here to explain.

Reply:
There isn't much to explain other than that (what I've observed) for VarDecl's, the variable name follows the "used" keyword. How should I explain this in a comment?

Reply:
I think a comment that shows the structure of a VarDecl here would make this clear, yes.